cpp/src/interop/dlpack.cpp

```diff
@@ -216,7 +220,7 @@ DLManagedTensor* to_dlpack(table_view const& input,
 {
   auto const num_rows = input.num_rows();
   auto const num_cols = input.num_columns();
-  if (num_rows == 0) { return nullptr; }
+  if (num_rows == 0 && num_cols == 0) { return nullptr; }
```

Comment:
If we have columns with no rows, then we can actually construct a type-correct dlpack tensor. This change enables the following to round-trip correctly:
```
df1 = cudf.DataFrame(cupy.empty((0, 3), dtype=float)
df2 = cudf.io.dlpack.from_dlpack(df1.to_dlpack())

assert_eq(df1, df2) => True
```

Again, if we have no columns there's no way to construct a type correct dlpack tensor (since the type information lives on the columns). In general cudf (possibly reasonably) throws away type information whenever a dataframe contains no columns, so I don't think we can easily do better here.